tests/unit/processor/requester/test_requester.py

```diff
@@ -0,0 +1,274 @@
+                "url": "http://mock-mock/",
+                "method": "GET",
+                "target_field_mapping": {"key1.key2.key3": "result.custom"},
+            },
+        },
+        {"message": "the message"},
+        {"message": "the message", "result": {"custom": "value"}},
+        {
+            "method": "GET",
+            "url": "http://mock-mock/",
+            "json": {"key1": {"key2": {"key3": "value"}}},
+            "content_type": "text/plain",
+            "status": 200,
+        },
+    ),
+]  # testcase, rule, event, expected, response
+
+failure_test_cases = [
+    (
+        "handles HTTPError",
+        {"filter": "message", "requester": {"url": "http://failure_mock", "method": "GET"}},
+        {"message": "the message"},
+        {"message": "the message", "tags": ["_requester_failure"]},
+        {
+            "method": "GET",
+            "url": "http://failure_mock",
+            "body": HTTPError("404"),
+            "content_type": "text/plain",
+            "status": 404,
+        },
+    ),
+    (
+        "timout error",
+        {
+            "filter": "message",
+            "requester": {"url": "http://failure_mock", "method": "GET", "timeout": 0.2},
+        },
+        {"message": "the message"},
+        {"message": "the message", "tags": ["_requester_failure"]},
+        {
+            "method": "GET",
+            "url": "http://failure_mock",
+            "body": ConnectTimeout(),
+            "content_type": "text/plain",
+            "status": 200,
+        },
+    ),
+    (
+        "does not overwrite if not permitted",
+        {
+            "filter": "message",
+            "requester": {"url": "http://failure_mock", "method": "GET", "target_field": "result"},
+        },
+        {"message": "the message", "result": "preexisting"},
+        {"message": "the message", "result": "preexisting", "tags": ["_requester_failure"]},
+        {
+            "method": "GET",
+            "url": "http://failure_mock",
+            "body": "the body",
+            "content_type": "text/plain",
+            "status": 200,
+        },
+    ),
+    (
+        "errors on missing fields",
+        {
+            "filter": "message",
+            "requester": {"url": "http://${missingfield}", "method": "GET"},
+        },
+        {"message": "the message"},
+        {"message": "the message", "tags": ["_requester_failure"]},
+        {},
+    ),
+]  # testcase, rule, event, expected, mock
+
+
+class TestRequester(BaseProcessorTestCase):
+
+    CONFIG: dict = {
+        "type": "requester",
+        "specific_rules": ["tests/testdata/unit/requester/specific_rules"],
+        "generic_rules": ["tests/testdata/unit/requester/generic_rules"],
+    }
+
+    @responses.activate
+    @pytest.mark.parametrize("testcase, rule, event, expected, response_kwargs", test_cases)
+    def test_testcases(
+        self, testcase, rule, event, expected, response_kwargs
+    ):  # pylint: disable=unused-argument
```

Comment:
If you use the `testcase` as a argument for the `assert` statement you can get rid of this pylint disable.  See line 263.

Reply:
done